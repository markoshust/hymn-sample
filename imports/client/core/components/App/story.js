import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import App from './App';

const stories = storiesOf('core/App', module);
const Dummy = () => (
  <div>{text('Content', 'This is a component')}</div>
);

stories.add('Text Child', () => (
  <App>{text('Content', 'This is text')}</App>
));

stories.add('Component Child', () => (
  <App>
    <Dummy />
  </App>
));
