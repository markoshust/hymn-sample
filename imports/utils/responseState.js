const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
