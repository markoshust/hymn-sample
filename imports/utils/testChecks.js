import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
