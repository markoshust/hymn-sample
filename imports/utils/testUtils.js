import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
