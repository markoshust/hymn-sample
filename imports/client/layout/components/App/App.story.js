import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import App from './App';

const stories = storiesOf('App', module);
const Dummy = () => (
  <div>{text('Content', 'This is a component')}</div>
);

stories.add('Text Element', () => (
  <App>{text('Content', 'This is text')}</App>
));

stories.add('Component', () => (
  <App>
    <Dummy />
  </App>
));
