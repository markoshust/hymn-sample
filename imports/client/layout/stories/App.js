import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import App from '../components/App';

const stories = storiesOf('App', module);

stories.add('Text Element', () => (
  <App>{text('Content', 'Hello World')}</App>
));

stories.add('Component', () => (
  <App>
    <div>{text('Content', 'Hello World')}</div>
  </App>
));
