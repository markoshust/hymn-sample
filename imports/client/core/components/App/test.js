import React from 'react';
import { renderCheck } from 'utils/testChecks';
import App from './App';

describe('<App />', () => {
  it('renders correctly', () => {
    renderCheck(
      <App>
        <div>Test</div>
      </App>
    );
  });
});
