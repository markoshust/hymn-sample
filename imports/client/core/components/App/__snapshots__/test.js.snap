exports[`<App /> renders correctly 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;
